src/groupeddataframe/splitapplycombine.jl

```diff
@@ -523,58 +541,66 @@ function _combine(gd::GroupedDataFrame,
     # and if a given column can be replaced in the future
     seen_cols = Dict{Symbol, Tuple{Bool, Int}}()
 
+    tasks = similar(cs_norm, Task)
+
     parentdf = parent(gd)
-    for i in eachindex(cs_norm, optional_transform)
+    for i in eachindex(cs_norm, optional_transform, tasks)
```

Comment:
if you agree with comment https://github.com/JuliaData/DataFrames.jl/pull/2574/files#r537099264 then `@sync` should be added here and below we should just call the post-processing anonymous function when everything is done being executed (so that we are sure that `idx_agg` after this loop and before next steps is properly set).

Reply:
Actually `@sync` isn't needed here since we call `fetch` on each task below (and we have to since we need to store tasks until they return).

Reply:
It is needed I think because you have to make sure that `idx_agg` HAS been computed before finishing this loop. Otherwise it is possible that you would start post-processing and `idx_agg` there would not be computed yet in this loop. So:
1) post processing would think that `idx_agg` is `nothing`
2) later actual processing would compute it

Because of this using `fetch` sequentially is not enough. You have to be sure you `fetch`ed everything before moving forward.

However - let us both review my line of reasoning again (it is for sure safe, but maybe in our actual case it is overly safe and the `idx_agg` issue will not hit us).

I am raising this because my experience is that these kinds of bugs (if I am right and there is a bug - I might be wrong) are heisenbugs (as if you start monitoring the process to diagnose them they disappear because main loop has enough time to finish its operations).

Reply:
OK, let's add it, anyway it shouldn't cost anything and it makes the code clearer.